### ISSUE TITLE
Do not raise S138, S1541, S3776 on top level statements

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#5624]
## Description

Currently, local functions are taken into consideration as part of the parent method in the following rules:
- S138 - Functions should not have too many lines of code
- S1541 - Functions should not be too complex
- S3776 - Cognitive Complexity of functions should not be too high

[Top level statements](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/top-level-statements) are used in more and more applications, where local functions appear as normal functions.

To reduce noise, we should just not raise these rules on top level statements. After #2470 gets defined for non-static local functions (which would require more specification and implementation effort) **we should consider creating a specific rule for top-level statements (which would not be confusing), when non-static local functions are abused.**

**Comment by andrei-epure-sonarsource** [comment]
(I'd like to have some feedback on this before working on it... #5625 should cover most scenarios where S3776 is obviously wrong on top level statements)

**Comment by costin-zaharia-sonarsource** [comment]
@andrei-epure-sonarsource regarding the feedback you need, do you plan to do something about it?

**Comment by andrei-epure-sonarsource** [comment]
I added a card on [our trello](https://trello.com/c/1GtKjMl3/758-take-decisions-regarding-the-support-of-local-functions-for-cognitive-complexity-and-other-types-of-complexity) (but I put it on pause due to lack of bandwidth on my side, anyone can pick it)

**Comment by andrei-epure-sonarsource** [comment]
Close as won't fix. The main problem were static local functions. We don't raise on static local functions, that reduced the noise.
